### ISSUE TITLE
[Bug] ReorderOperation with more than 200 records

### ISSUE THREAD

**Issue by o15a3d4l11s2** [Laravel-Backpack/CRUD#3722]
# Bug report

### What I did

I set up a model that uses `ReorderOperation`.

### What I expected to happen

Reordering and saving stores the information in the database and after refresh of the view, the data is still shown with the new order.

### What happened

Order was not persisted into the database. After a refresh, the old order was shown.

### What is the cause of the issue

When there are more than 200 records to reorder (not an edge case with a bigger shop with multiple subcategories), the `$.ajax` request data gets cut inside PHP and PHP receives only the first 200 records. The reason is that data is sent as an array from JS and PHP has a `max_input_vars` which is by default 1000. Every record in the array (every model item) contains multiple properties (`{item_id: "1", parent_id: 5, depth: 0, left: 11 right: 21}`), so 200 multiplied by 5.

### What I've already tried to fix it

There are two possible solutions:

1. Increase the `max_input_vars` value. I do not like this because of the possible denial of service concerns and changing the PHP ini is not always easy for all hostings.
2. Instead of sending an array, send the data as json string and decode it back when received in PHP. This requires two changes in the code:
   1. In `vendor/backpack/crud/src/resources/views/crud/reorder.blade.php` replace

        ```php
        $.ajax({
            url: '{{ url(Request::path()) }}',
            type: 'POST',
            data: { tree: arraied },
        })
        ```

        with

        ```php
        $.ajax({
            url: '{{ url(Request::path()) }}',
            type: 'POST',
            data: { tree: JSON.stringify(arraied) },
        })
        ```

    2. In `vendor/backpack/crud/src/app/Http/Controllers/Operations/ReorderOperation.php` replace

        ```php
        public function saveReorder()
        {
            $this->crud->hasAccessOrFail('reorder');

            $all_entries =\Request::input('tree');

            if (count($all_entries)) {
                $count = $this->crud->updateTreeOrder($all_entries);
            } else {
                return false;
            }

            return 'success for '.$count.' items';
        }
        ```

        with

        ```php
        public function saveReorder()
        {
            $this->crud->hasAccessOrFail('reorder');

            $all_entries = json_decode(\Request::input('tree'), true);

            if (count($all_entries)) {
                $count = $this->crud->updateTreeOrder($all_entries);
            } else {
                return false;
            }

            return 'success for '.$count.' items';
        }
        ```

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

#### PHP VERSION:
PHP 8.0.5 (cli) (built: May  3 2021 11:30:57) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.5, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.5, Copyright (c), by Zend Technologies

#### LARAVEL VERSION:
v8.40.0@a654897ad7f97aea9d7ef292803939798c4a02a4

#### BACKPACK VERSION:
4.1.43@973e8ab33d42f02272f65e9688ea250fc9502d3d

**Comment by bedoz** [comment]
Nice idea @o15a3d4l11s2, i had also this issue time ago, that we solved with max input vars of course, but yours second one is better than mine.

**Comment by o15a3d4l11s2** [comment]
@bedoz, that was also my first reaction.
I hope Pedro/Cristian/António will agree with the second idea, because it will solve possible future issues for other users too. 

Thanks for confirming the issue.

**Comment by tabacitu** [comment]
Totally agree, let's do that @o15a3d4l11s2 👏👏👏  Very elegant. I'd be cautious and categorize this as a breaking change, though, and release it in 4.2 (soon enough). That's because people might have already customized only the view, or only the controller method, and for those people we'd be introducing a breaking change...

Would you like to submit a PR with this, or would you like us to do it @o15a3d4l11s2 ?

**Comment by o15a3d4l11s2** [comment]
I agree this is a breaking change, @tabacitu. I can submit a PR in the next 2 days.

**Comment by tabacitu** [comment]
Closing. Was fixed in https://github.com/Laravel-Backpack/CRUD/pull/3808 which was merged into 4.2

**Comment by Innovicont** [comment]
Thanks! It's work, you save my time...
